arm9/src/gf_gfx_loader.c

```diff
@@ -0,0 +1,350 @@
+                break;
+            case NNS_G2D_VRAM_TYPE_2DSUB:
+                pCharacterData->mapingType = (GXOBJVRamModeChar)(reg_GXS_DB_DISPCNT & (REG_GXS_DB_DISPCNT_EXOBJ_MASK | REG_GXS_DB_DISPCNT_OBJMAP_CH_MASK));
+                break;
+            default:
+                ;
+            }
+            load_funcs[whichRoutine](pCharacterData, baseAddr, type, pImageProxy);
+        }
+        FreeToHeap(pFile);
+    }
+}
+
+THUMB_FUNC void * GfGfxLoader_GetCharData(NarcId narcId, s32 memberId, BOOL isCompressed, NNSG2dCharacterData ** ppCharData, u32 heap_id)
+{
+    void * pFile = GfGfxLoader_LoadFromNarc(narcId, memberId, isCompressed, heap_id, FALSE);
+    if (pFile != NULL)
+    {
+        if (!NNS_G2dGetUnpackedBGCharacterData(pFile, ppCharData))
+        {
+            FreeToHeap(pFile);
+            return NULL;
+        }
+    }
+    return pFile;
+}
+
+THUMB_FUNC void * GfGfxLoader_GetScrnData(NarcId narcId, s32 memberId, BOOL isCompressed, NNSG2dScreenData ** ppScrData, u32 heap_id)
+{
+    void * pFile = GfGfxLoader_LoadFromNarc(narcId, memberId, isCompressed, heap_id, FALSE);
+    if (pFile != NULL)
+    {
+        if (!NNS_G2dGetUnpackedScreenData(pFile, ppScrData))
+        {
+            FreeToHeap(pFile);
+            return NULL;
+        }
+    }
+    return pFile;
+}
+
+THUMB_FUNC void * GfGfxLoader_GetPlttData(NarcId narcId, s32 memberId, NNSG2dPaletteData ** ppPltData, u32 heap_id)
+{
+    void * pFile = GfGfxLoader_LoadFromNarc(narcId, memberId, FALSE, heap_id, FALSE);
+    if (pFile != NULL)
+    {
+        if (!NNS_G2dGetUnpackedPaletteData(pFile, ppPltData))
+        {
+            FreeToHeap(pFile);
+            return NULL;
+        }
+    }
+    return pFile;
+}
+
+THUMB_FUNC void * GfGfxLoader_GetCellBank(NarcId narcId, s32 memberId, BOOL isCompressed, NNSG2dCellDataBank ** ppCellBank, u32 heap_id)
+{
+    void * pFile = GfGfxLoader_LoadFromNarc(narcId, memberId, isCompressed, heap_id, FALSE);
+    if (pFile != NULL)
+    {
+        if (!NNS_G2dGetUnpackedCellBank(pFile, ppCellBank))
+        {
+            FreeToHeap(pFile);
+            return NULL;
+        }
+    }
+    return pFile;
+}
+
+THUMB_FUNC void * GfGfxLoader_GetAnimBank(NarcId narcId, s32 memberId, BOOL isCompressed, NNSG2dAnimBankData ** ppAnimBank, u32 heap_id)
+{
+    void * pFile = GfGfxLoader_LoadFromNarc(narcId, memberId, isCompressed, heap_id, FALSE);
+    if (pFile != NULL)
+    {
+        if (!NNS_G2dGetUnpackedAnimBank(pFile, ppAnimBank))
+        {
+            FreeToHeap(pFile);
+            return NULL;
+        }
+    }
+    return pFile;
+}
+
+THUMB_FUNC void * GfGfxLoader_UncompressFromNarc(NarcId narcId, s32 memberId, u32 heap_id)
+{
+    return GfGfxLoader_LoadFromNarc(narcId, memberId, TRUE, heap_id, FALSE);
+}
+
+THUMB_FUNC void * GfGfxLoader_LoadFromNarc(NarcId narcId, s32 memberNo, BOOL isCompressed, u32 heap_id, BOOL allocAtEnd)
+{
+    void * dest;
+    void * dest2;
+    if (isCompressed || allocAtEnd == TRUE)

```

Comment:
== TRUE? is this needed for matching?

Reply:
Yes, without question. If the question related to `!= FALSE` it'd be a different question.